pipeline/archive/archive.go

```diff
@@ -97,7 +97,7 @@ func create(ctx *context.Context, platform string, groups map[string][]context.B
 				return fmt.Errorf("failed to add %s -> %s to the archive: %s", binary.Path, binary.Name, err.Error())
 			}
 		}
-		ctx.AddArtifact(archivePath)
+		ctx.AddArtifactFromFile(archivePath)
```

Comment:
If `AddArtifactFromFile` doesn't need additional context you could have an `NewArtifactFromFile` constructor for an `Artifact`.